src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -53,3 +53,26 @@
      handle-connection)))
 
 (rf/reg-event-fx :wallet/get-key-pair-export-connection get-key-pair-export-connection)
+
+(defn wallet-validate-seed-phrase
+  [_cofx [seed-phrase on-success on-error]]
+  {:fx [[:multiaccount/validate-mnemonic [seed-phrase on-success on-error]]]})
+
+(rf/reg-event-fx :wallet/validate-seed-phrase wallet-validate-seed-phrase)
```

Comment:
Here we're creating a simple event handler for calling the `:multiaccount/validate-mnemonic` effect. This effect should double check that the provided seed-phrase is an actual seed-phrase associated to the user.

Reply:
The definition for this effect might already exist, if it does please lmk 🙏

Reply:
IMO, creating a re-frame event to to dispatch an effect is not needed, so it's not needed to make it general.

We already have a similar usage in:
`status-im.contexts.wallet.add-account.create-account.events`
named:
`:wallet/seed-phrase-entered`

And I considered the same as you: to create a general event that receives the params and callbacks, but IMO it's like doing this:

```clojure
(defn join-strings [& strs]
  (apply str strs))
```
It helps a little, but we could use `str` directly and it's not bad, also I feel it like adding an extra layer that makes the code less simple.

That's my take on this topic.

Reply:
Yeah I think that's a fair perspective 👍 

I suppose I was weighing my choices between creating a generic event for calling an effect, or creating a bespoke event for calling the effect. I'm not sure what's the recommended approach, but I was hoping to avoid calling the side effect from my view and only dispatch an event. 

But I'm not sure if that view should have it's own `:seed-phrase-entered` event or if I should re-use the `:wallet/seed-phrase-entered` event. The `:wallet/seed-phrase-entered` seems to be tied directly to the wallet logic, so maybe we can refactor that so it can be shared?

Reply:
The existing one is tied to the logic because in re-frame we aren't expecting creating events only to dispatch effects.

So if you create a new one only for this view I think it's completely ok :+1: if you decide to abstract them in a single one, I also think is ok as long as they share some logic besides dispatching the effect.

Consider that dispatching an effect  is just "the re-frame way" of invoking a function with side effects. 
Wrapping an effect in an event makes sense when we will always perform some logic before dispatching the effect, but this logic should need something from `DB` otherwhise this logic could live inside the effect handler and, again, we wouldn't need a wrapper for it.